### ISSUE TITLE
[ML] Memory access violation in testMultiClassClassificationFeatureImportanceAllShap

### ISSUE THREAD

**Issue by valeriy42** [elastic/ml-cpp#1080]
There seems to be some none-deterministic memory violation in `CDataFrameAnalyzerFeatureImportanceTest/testMultiClassClassificationFeatureImportanceAllShap` (at least on Windows):
```
2020-03-24 15:23:40,393184 UTC [2260] DEBUG CTestObserver.cc@23 +------------------------------------------------------------------------------------------------+
2020-03-24 15:23:40,393184 UTC [2260] DEBUG CTestObserver.cc@24 |  CDataFrameAnalyzerFeatureImportanceTest/testMultiClassClassificationFeatureImportanceAllShap  |
2020-03-24 15:23:40,393184 UTC [2260] DEBUG CTestObserver.cc@25 +------------------------------------------------------------------------------------------------+
2020-03-24 15:23:40,405184 UTC [2260] DEBUG CDataFrameAnalyzer.cc@102 Received 2000 rows
2020-03-24 15:23:41,373235 UTC [2260] INFO  CBoostedTreeImpl.cc@1193 loss* = 0.0247686, regularization* = (depth penalty multiplier = 2.000000, soft depth limit = 5.000000, soft depth tolerance = 0.100000, tree size penalty multiplier = 10.000000, leaf weight penalty multiplier = 1.000000), downsample factor* = 0.39685, eta* = 0.9, eta growth rate per tree* = 1.05, maximum number trees* = 2, feature bag fraction* = 1
2020-03-24 15:23:41,397236 UTC [2260] INFO  CBoostedTreeImpl.cc@260 Training finished after 2 iterations. Time per iteration in ms mean: 114.333 std. dev:  62.9312
unknown location(0): fatal error: in "CDataFrameAnalyzerFeatureImportanceTest/testMultiClassClassificationFeatureImportanceAllShap": memory access violation occurred at address 0xffffffff, while attempting to  read inaccessible data
```

**Comment by droberts195** [comment]
https://elasticsearch-ci.elastic.co/job/elastic+machine-learning+7.x+windows/251/console is an example of this happening in a public CI build.

**Comment by valeriy42** [comment]
Was fixed in #1086
